client/extensions/woocommerce/app/order/order-activity-log/index.js

```diff
@@ -39,21 +39,30 @@ function getSortedEvents( events ) {
 	return eventsByDay;
 }
 
-class OrderNotes extends Component {
+class OrderActivityLog extends Component {
 	static propTypes = {
 		orderId: PropTypes.number.isRequired,
 		siteId: PropTypes.number.isRequired,
 	};
 
 	constructor( props ) {
 		super( props );
-		this.state = {
-			openIndex: 0,
-		};
 	}
 
-	toggleOpenDay = index => {
-		this.setState( () => ( { openIndex: index } ) );
+	componentWillMount() {
+		this.setState( { openDay: last( keys( this.props.eventsByDay ) ) } );
+	}
+
+	componentWillReceiveProps( props ) {
+		const newOpenDay = last( keys( props.eventsByDay ) );
+		//new day has been appended, open it
+		if ( ! this.props.eventsByDay[ newOpenDay ] ) {
```

Comment:
This is saying we have a new latest day in the incoming props, but it doesn't exist in the last prop set, and therefor it's a new day, so we "open" it - right? It's a little unclear, maybe update your comment/variable names? I try to use `nextProps` for `componentWillReceiveProps` just to be clear about the timeline.